Clues for admins to edit user profiles, accounts

weedySeaDragon opened the pull request:
## PT Story: A way to edit a user from viewing a user is missing as an admin
#### PT URL: https://www.pivotaltracker.com/story/show/169760526

Provide clues so admins can edit user information: make links to editting user profiles and user accounts more obvious.

- I've put in some visual cues and words.  Are they helpful?  Suggestions?  See the screenshots below.
- I picked an icon for the user profile and for the user account.  Are these ok? 
- I don't really know what to use for the user account.  A _user account_ shows  membership + application + payment status).  I didn't see a FontAwesome icon that represents those well.
- We don't have to use a FontAwesome icon; we can use any image we want.  I just picked from FontAwesome to keep it easy. (We could use a tiny version of on of the SHF graphics if it'd show up well enough.)


## Changes proposed in this pull request:
1.  Small changes to UI to help provide clues. (See screenshots below)

## Screenshots:


### 1. ~~Added words to column headers to explain links;~~ added tooltips to each entry
Reverted the headers.  (Tooltips remain)

_~~Added words to 2 column headers to explain where you'll go when you click on an entry.~~ (And also added tooltips to each entry so when you hover you'll see something like "Edit account for user Firstname Lastname"_


<img width="883" alt="allUsers-headers-reverted" src="https://user-images.githubusercontent.com/673794/71744473-31577a80-2e1c-11ea-901d-4888b0739a6a.png">

---

### 2. Edit Profile page: Added icon in header to go to the _view_ account
_Added an icon to the Header that links to the (view) account page._

<img width="798" alt="edit-profile-icon-to-view-acct" src="https://user-images.githubusercontent.com/673794/71744472-31577a80-2e1c-11ea-8659-53f6206a1cdc.png">

---

### 3(a). View Account page: Added **_edit_** icon (the pencil thing) in header to go to the _edit_ account page.  Also has the icon to go to the Edit Profile page.
_Added an "edit" icon to the Header that links to the edit account page._

<img width="833" alt="view-acct-icons-edit-profile" src="https://user-images.githubusercontent.com/673794/71744471-31577a80-2e1c-11ea-895f-ae045d4ed4a9.png">

---

### 3. Edit Account page: Added icon in header to go to the edit profile page.
_Added an icon to the Header that links to the edit profile page._

<img width="791" alt="edit-acct-icon-to-profile" src="https://user-images.githubusercontent.com/673794/71744470-31577a80-2e1c-11ea-97be-965ae2837208.png">



---
### 4.  User Account: Application section - add links, show all companies

- Show all companies for an application. Link each company #, name to the company
- Linked the application state to the application. 
- Used i18n for the "User Info" section title

<img width="775" alt="many-companies-links-AdminTitle-localed" src="https://user-images.githubusercontent.com/673794/72093733-d74a2f80-32c9-11ea-9bcd-bb4cd21f8007.png">

---







---
## Ready for review:
@thesuss

weedySeaDragon committed:
account: link application

weedySeaDragon committed:
profile: h1: admin icon links to edit account

weedySeaDragon committed:
acct: profile: h1: admin icon links to edit profile; i18n admin User Info title

weedySeaDragon committed:
all users: column headers, tooltips for [td]s

weedySeaDragon committed:
locale and stylesheet changes

weedySeaDragon committed:
Merge remote-tracking branch 'AgileVentures/develop' into sprint83-admin-can-edit-all-user-info

weedySeaDragon committed:
helpers for standardized icons to use

weedySeaDragon committed:
use standardized icon (via helper methods)

thesuss commented on config/locales/sv.yml:
```diff
@@ -893,6 +895,8 @@ sv:
       edit_member_status: &edit_member_status Redigera medlemsstatus
       pay_membership: &pay_membership Betala medlemsavgift
+      edit_user_profile: &edit_user_profile Edit the profile for %{name}
+      admin_user_info_title: Användare Info
```
Information om användaren

thesuss commented on config/locales/sv.yml:
```diff
@@ -936,10 +942,16 @@ sv:
       applications_tooltip: Användaren har en öppen ansökan
       expire_date: *expire_date
       member_packet: *member_packet
```
pkt

weedySeaDragon replied:
the "*" means that it's a reference to the actual entry.  (It's defined on line 881.  The "&" in _that_ entry means that it is the key definition.
In the `sv file it is "pkt"  .  But... in the `en` file it was "member packet".   I fixed it.
Good eye!

thesuss commented on config/locales/sv.yml:
```diff
@@ -936,10 +942,16 @@ sv:
       applications_tooltip: Användaren har en öppen ansökan
       expire_date: *expire_date
       member_packet: *member_packet
+      edit_profile: Edit Profile
```
not needed

thesuss commented on config/locales/sv.yml:
```diff
@@ -936,10 +942,16 @@ sv:
       expire_date: *expire_date
       member_packet: *member_packet
+      edit_profile: Edit Profile
+      edit_account: (Edit Acct.)
```
not needed

thesuss commented:
The icon by the very red header makes no sense... 
Use a button at the bottom like for company or application? 

As for the user table, short info, short headline. 
As for the swedish version that was as it was :)

weedySeaDragon committed:
refactor SetAppConfiguration into own concern so it can be used elsewhere

weedySeaDragon committed:
refactor large 'show' view into partials

weedySeaDragon committed:
locales: reorganize to match partials created; add section for admin_only:user_account

weedySeaDragon committed:
update t() entries for re-organized locales

weedySeaDragon committed:
add edit icon to 'standard' icon helper methods

weedySeaDragon committed:
view, controller action, routes for Admin only: edit user account

weedySeaDragon committed:
use 'shf-icon' CSS class to style icons on headings

weedySeaDragon committed:
add feature: admin edits user account (and path helper for the steps)

weedySeaDragon committed:
update locale entries

weedySeaDragon commented:
_@thesuss said:_
> The icon by the very red header makes no sense...
> Use a button at the bottom like for company or application?

I thought it might be a quick way for the admin to switch back and forth between the User Account and User Profile for the user. 
I think maybe the icons could be helpful.  I popped them in there with the header because I couldn't quickly come up with a different place, and just wanted some feedback.

I also added an Edit icon so that if you (as an admin) are _viewing_ a user profile, you can click on the "edit" icon and quickly switch to the _edit_ view.  
I'll update the screenshots so you can see.  And if you don't like the icons at all, no worries.  We can take them out.

(And let me know what color to change the "very red header" to and I'll make it so. :-)

> As for the user table, short info, short headline.
> As for the swedish version that was as it was :)
- fixed.

weedySeaDragon committed:
Users list (index): revert header changes (keep them short)

weedySeaDragon committed:
don't use @ inst. vars in partials. pass in locals instead

weedySeaDragon committed:
remove cruft

weedySeaDragon committed:
added comments; renamed file so the purpose is clear

weedySeaDragon committed:
add description to show this covers more than just admins

weedySeaDragon committed:
locales: added references to existing entries for shf_application_min_info_as_table partial

weedySeaDragon committed:
User Acct - Application: styles for Org Nr. and Name table heading ;+ comment

weedySeaDragon committed:
User Acct- Application section: Show all companies for an application
